framework/test/unit/modules/auth/command.spec.ts

```diff
@@ -0,0 +1,480 @@
+				.createTransactionContext({
+					transaction: new Transaction({ ...transaction.toObject(), params }),
+					networkIdentifier,
+				})
+				.createCommandVerifyContext(
+					registerMultisignatureParamsSchema,
+				) as CommandVerifyContext<RegisterMultisignatureParams>;
+			const result = await registerMultisignatureCommand.verify(context);
+			expect(result.error?.message).toBe(
+				'The numberOfSignatures is bigger than the count of Mandatory and Optional keys.',
+			);
+		});
+
+		it('should return error when numberOfSignatures is smaller than mandatory key count', async () => {
+			const params = codec.encode(registerMultisignatureParamsSchema, {
+				...decodedParams,
+				numberOfSignatures: 1,
+			});
+			const context = testing
+				.createTransactionContext({
+					transaction: new Transaction({ ...transaction.toObject(), params }),
+					networkIdentifier,
+				})
+				.createCommandVerifyContext(
+					registerMultisignatureParamsSchema,
+				) as CommandVerifyContext<RegisterMultisignatureParams>;
+			const result = await registerMultisignatureCommand.verify(context);
+			expect(result.error?.message).toBe(
+				'The numberOfSignatures needs to be equal or bigger than the number of Mandatory keys.',
+			);
+		});
+
+		it('should return error when mandatory and optional key sets are not disjointed', async () => {
+			const params = codec.encode(registerMultisignatureParamsSchema, {
+				numberOfSignatures: 2,
+				mandatoryKeys: [
+					Buffer.from('48e041ae61a32777c899c1f1b0a9588bdfe939030613277a39556518cc66d371', 'hex'),
+					Buffer.from('483077a8b23208f2fd85dacec0fbb0b590befea0a1fcd76a5b43f33063aaa180', 'hex'),
+				],
+				optionalKeys: [
+					Buffer.from('483077a8b23208f2fd85dacec0fbb0b590befea0a1fcd76a5b43f33063aaa180', 'hex'),
+				],
+			});
+			const context = testing
+				.createTransactionContext({
+					transaction: new Transaction({ ...transaction.toObject(), params }),
+					networkIdentifier,
+				})
+				.createCommandVerifyContext(
+					registerMultisignatureParamsSchema,
+				) as CommandVerifyContext<RegisterMultisignatureParams>;
+
+			const result = await registerMultisignatureCommand.verify(context);
+			expect(result.error?.message).toBe(
+				'Invalid combination of Mandatory and Optional keys. Repeated keys across Mandatory and Optional were found.',
+			);
+		});
+
+		it('should return error when mandatory keys set is not sorted', async () => {
+			const params = codec.encode(registerMultisignatureParamsSchema, {
+				...decodedParams,
+				numberOfSignatures: 2,
+				mandatoryKeys: [
+					Buffer.from('48e041ae61a32777c899c1f1b0a9588bdfe939030613277a39556518cc66d371', 'hex'),
+					Buffer.from('483077a8b23208f2fd85dacec0fbb0b590befea0a1fcd76a5b43f33063aaa180', 'hex'),
+				],
+			});
+			const context = testing
+				.createTransactionContext({
+					transaction: new Transaction({ ...transaction.toObject(), params }),
+					networkIdentifier,
+				})
+				.createCommandVerifyContext(
```

Comment:
instead of casting `as` every time, it would be probably nice to update this function definition to be like
```
createCommandVerifyContext<T = Record<string, unknown>>(paramsSchema: Schema): CommandVerifyContext<T>
```
and call with 
```
.createCommandVerifyContext<RegisterMultisignatureParams>(registerMultisignatureParamsSchema);
```